app/controllers/search_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SearchController < ApplicationController
+  def index; end
+
+  def create
+    redirect_to recipes_path(params: search_params)
+  end
+
+  private
+
+  def search_params
+    params.permit(:ingredients,
+                  :time,
+                  :type,
+                  :vegetarian,
+                  :keto,
+                  'dairy free',
+                  'gluten free',
+                  :vegan)
```

Comment:
Soooooo many params.....

Reply:
There might be an easier way to produce 5 buttons, and have them all assigned to a single param (:diet) but I haven't messed around with rails form helpers that much.